arangod/Aql/OptimizerRules.cpp

```diff
@@ -8348,6 +8348,11 @@ void arangodb::aql::parallelizeGatherRule(Optimizer* opt,
                                           OptimizerRule const& rule) {
   TRI_ASSERT(ServerState::instance()->isCoordinator());
 
+  if (plan->getAst()->query().vocbase().isOneShard()) {
```

Comment:
Does it hurt if it is applied? Or is this "just" unnecessary work?

If yes:
If we  are in a sharded database "by accident" have collections with only 1 shard, that are all on the same server we may apply a plan that looks similar to a oneShardDB plan.